src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -123,9 +119,14 @@ object RuleUtils {
           entry.setTagValue(
             relation.plan,
             IndexLogEntryTags.HYBRIDSCAN_REQUIRED,
-            !(commonCnt == entry.sourceFileInfoSet.size && commonCnt == inputSourceFiles.size))
+            !(commonCnt == entry.sourceFileInfoSet.size
+              && commonCnt == relation.allFileInfos.size))
         }
         isCandidate
+      }) {
```

Comment:
`)` here is closing `if`, and for the following it's easy to consider `{` as the starting block of `if`.
```
if (entry.withCachedTag(relation.plan, IndexLogEntryTags.IS_HYBRIDSCAN_CANDIDATE) {`
```
Is there a better way to restructure?